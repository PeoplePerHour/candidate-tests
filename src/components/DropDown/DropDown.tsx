import * as React from "react";

interface IAppProps {
  onChange(val: string): void;
  onSelect?:void
  value?: string;
  option?: [];
  Options: string[];
  defaultText:string
  
}

const DropDown: React.FC<IAppProps> = props => {
  const { Options, value, onChange,defaultText } = props;

  const [selected, setSelected] = React.useState(null);
  const [opened, setOpened] = React.useState(false);

  React.useEffect(() => {
    setSelected(value);
  }, [value]);

  const onOpen = () => {
    setOpened(!opened);
  };

  const onSelect = (option:string) => {
    setOpened(false);
    setSelected(option);
    onChange(option);
  };

  return (
    <div className="select-dd" onClick={onOpen}>
      <span>{selected ? selected : defaultText}</span>
      {Options?
      <ul className={opened ? "show" : "hide"}>
        <Option key={4444} option={defaultText} onSelect={onSelect} />
        {Options.map((o:string, i:number) => (
          <Option key={i} option={o} onSelect={onSelect} />
        ))}
      </ul>:null}
    </div>
  );
};
const Option = (props:any) => {
  const { onSelect, option } = props;

  const handleSelect = (e:React.MouseEvent) => {
    e.preventDefault();
    onSelect(option);
  };

  return (
    <li>
      <a href="#" onClick={handleSelect}>
        {option}
      </a>
    </li>
  );
};

export default DropDown;
