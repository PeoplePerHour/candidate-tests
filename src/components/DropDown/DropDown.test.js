
import * as  React from 'react';
import * as ReactDOM from 'react-dom';

import DropDown from './DropDown'


it('renders DropDown without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<DropDown/>, div);
    expect(div.firstChild).toMatchSnapshot();
    ReactDOM.unmountComponentAtNode(div);
})
