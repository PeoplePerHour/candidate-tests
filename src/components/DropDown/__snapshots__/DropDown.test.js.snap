// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DropDown without crashing 1`] = `
<div
  class="select-dd"
>
  <span />
</div>
`;
